circle-mlir/circle-mlir/lib/pass/src/DumpCircleOpsPass.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __CIRCLE_MLIR_PASS_DUMP_CIRCLE_OPS_PASS_H__
+#define __CIRCLE_MLIR_PASS_DUMP_CIRCLE_OPS_PASS_H__
+
+#include <mlir/Dialect/Func/IR/FuncOps.h>
+#include <mlir/Dialect/Func/Transforms/Passes.h>
+
+#include <functional>
+
+namespace mlir
+{
+namespace Circle
+{
+
+struct DumpCircleOpsPass
+  : public mlir::PassWrapper<DumpCircleOpsPass, mlir::OperationPass<mlir::func::FuncOp>>
+{
+  DumpCircleOpsPass() = default;
+  DumpCircleOpsPass(const DumpCircleOpsPass &pass)
+    : mlir::PassWrapper<DumpCircleOpsPass, OperationPass<mlir::func::FuncOp>>()
+  {
+    _getOStream = pass._getOStream;
```

Comment:
(Optional) How about adding an assertion or an exception handling to prevent a `nullptr` from being passed? 
```suggestion
  assert(pass._getOStream != nullptr);
    _getOStream = pass._getOStream;
```

Reply:
I think `_getOStream` here can be `nullptr`.
there's setter `void ostream(GetOStream_t os)` which can update the value.